### DISCUSSION TITLE
where to code after the Hello Drone running, for RL implementation and Object Detection

### DISCUSSION THREAD

**Discussion by Roopesh-Bharatwaj-K-R** [microsoft/AirSim#3609] [Support Q&A] [answered]
### where to code after the Hello Drone running, for RL implementation and Object Detection 

Which files to implement codes for object detection and automatic drone pathing. 

### Include context on what you are trying to achieve
I have completed until the Hello Drone which runs on with the manual Control(external controller) and How i can make the drone movements autonomous and in which file do i need to code for the object detection Algorithm  .

**Comment by ahmed-elsaharti** [comment, marked as answer]
>where to code after the Hello Drone running, for RL implementation and Object Detection

Similar to #3550, If you're planning on using the [python APIs](https://microsoft.github.io/AirSim/apis/), all you would need to do is make sure the airsim package is installed through pip and you can start using the APIs similar to the [demo python scripts](https://github.com/microsoft/AirSim/tree/master/PythonClient/multirotor).

>How i can make the drone movements autonomous 

you can use the [moveToPositionAsync](https://github.com/microsoft/AirSim/blob/6d20663e0818448613be0332009b02bd6341589e/PythonClient/airsim/client.py#L982), the [moveByVelocityAsync](https://github.com/microsoft/AirSim/blob/6d20663e0818448613be0332009b02bd6341589e/PythonClient/airsim/client.py#L959) or any of the other movement API calls. For examples on how to use these, check the [demo files](https://github.com/microsoft/AirSim/tree/master/PythonClient/multirotor)

>which file do i need to code for the object detection Algorithm

You can create your own python script as long as you `import airsim`. Or you can modify any of the [demo python scripts](https://github.com/microsoft/AirSim/tree/master/PythonClient/multirotor)

**Comment by Roopesh-Bharatwaj-K-R** [comment]
Thank You very Much, really Helpful !
